src/common/transformations/tests/common_optimizations/convert_u4_weights_zero_point_to_scalar.cpp

```diff
@@ -0,0 +1,202 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/convert_u4_weights_zero_point_to_scalar.hpp"
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/ov_test_utils.hpp"
+#include "openvino/core/model.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/pass/manager.hpp"
+
+using namespace testing;
+using namespace ov;
+
+TEST_F(TransformationTestsF, ConvertU4WeightsFloatZeroPointToScalar) {
+    auto weights_precision = ov::element::u4;
+    auto decompression_precision = ov::element::f32;
+    ov::Shape weights_shape{32, 128, 64};
+    ov::Shape decompression_shape{32, 1, 64};
+    {
+        auto weights = ov::op::v0::Constant::create(weights_precision, weights_shape, {4});
+        auto convert = std::make_shared<ov::op::v0::Convert>(weights, decompression_precision);
+        auto zero_point = ov::op::v0::Constant::create(decompression_precision, decompression_shape, {8.1f});
+        auto subtract = std::make_shared<ov::op::v1::Subtract>(convert, zero_point);
+        auto scale = ov::op::v0::Constant::create(decompression_precision, decompression_shape, {3.f});
+        auto multiply = std::make_shared<ov::op::v1::Multiply>(subtract, scale);
+        model = std::make_shared<Model>(NodeVector{multiply}, ParameterVector{});
+        manager.register_pass<ov::pass::ConvertU4WeightsZeroPointToScalar>();
+    }
+    {
+        ov::Shape scalar_shape{};
+        auto weights = ov::op::v0::Constant::create(weights_precision, weights_shape, {4});
+        auto convert = std::make_shared<ov::op::v0::Convert>(weights, decompression_precision);
+        auto zero_point = ov::op::v0::Constant::create(decompression_precision, scalar_shape, {8.1f});
+        auto subtract = std::make_shared<ov::op::v1::Subtract>(convert, zero_point);
+        auto scale = ov::op::v0::Constant::create(decompression_precision, decompression_shape, {3.f});
+        auto multiply = std::make_shared<ov::op::v1::Multiply>(subtract, scale);
+        model_ref = std::make_shared<Model>(NodeVector{multiply}, ParameterVector{});
+    }
```

Comment:
You can consider enabling for positive tests:
```
    comparator.enable(FunctionsComparator::CmpValues::ACCURACY);
    comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
```

Reply:
It won't be superfluous. Thanks, added